lib/identifiers/pubmed_id.rb

```diff
@@ -1,7 +1,9 @@
 module Identifiers
   class PubmedId
     def self.extract(str)
-      str.to_s.scan(/(?<=^|[[:space:]])0*(?!0)(\d+)(?=$|[[:space:]])/).flatten
+      str = str.to_s
+      str.scan(/(?<=^|[[:space:]])0*(?!0)(\d+)(?=$|[[:space:]])/).flatten |
```

Comment:
Perhaps we could extract these patterns into named constants to explain the two types we support?

```ruby
ZERO_PADDED_NUMBER = ...
PUBMED_URL = ...
```